tools/themis/src/rules.rs

```diff
@@ -0,0 +1,329 @@
+use std::collections::HashMap;
+
+use super::{style, utils, Error, Expected, PackageOutcome, Workspace};
+
+/// ensure all crates have the `publish = <true/false>` specification
+pub fn has_publish_spec(workspace: &Workspace) -> Result<(), Error> {
+    let outliers: Vec<_> = workspace
+        .members
+        .iter()
+        .filter(|pkg| pkg.raw["package"].get("publish").is_none())
+        .map(|pkg| PackageOutcome { pkg, value: None })
+        .collect();
+
+    if !outliers.is_empty() {
+        return Err(Error::OutcomeError {
+            msg: "These packages should have the `publish` specification".to_string(),
+            expected: Some(Expected { value: "publish = <true/false>".to_string(), reason: None }),
+            outliers,
+        });
+    }
+
+    Ok(())
+}
+
+/// ensure all crates specify a MSRV
+pub fn has_rust_version(workspace: &Workspace) -> Result<(), Error> {
+    let outliers: Vec<_> = workspace
+        .members
+        .iter()
+        .filter(|pkg| pkg.raw["package"].get("rust-version").is_none())
+        .map(|pkg| PackageOutcome { pkg, value: None })
+        .collect();
+
+    if !outliers.is_empty() {
+        return Err(Error::OutcomeError {
+            msg: "These packages should specify a Minimum Supported Rust Version (MSRV)"
+                .to_string(),
+            expected: None,
+            outliers,
+        });
+    }
+
+    Ok(())
+}
+
+/// ensure all crates are versioned to v0.0.0
+pub fn is_unversioned(workspace: &Workspace) -> Result<(), Error> {
+    let outliers = workspace
+        .members
+        .iter()
+        .filter(|pkg| pkg.parsed.version != semver::Version::new(0, 0, 0))
+        .map(|pkg| PackageOutcome { pkg, value: Some(pkg.parsed.version.to_string()) })
+        .collect::<Vec<_>>();
+
+    if !outliers.is_empty() {
+        return Err(Error::OutcomeError {
+            msg: "These packages shouldn't be versioned".to_string(),
+            expected: Some(Expected { value: "0.0.0".to_string(), reason: None }),
+            outliers,
+        });
+    }
+
+    Ok(())
+}
```

Comment:
Oh, ignore my other comment. I thought it would be better, to rewrite this check, to make sure all versions are the same. 
This works too.